### PR TITLE
Add support for Dependabot alert endpoints

### PR DESCRIPTION
**Pull request by francois2metz**
This add support for reading dependabot alerts: https://docs.github.com/en/rest/dependabot/alerts

I added the `First` and `Last` parameters to the `ListCursorOptions`. It seems to replace the `PerPage` parameter.

### CODE REVIEW

File: github/dependabot_alerts.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+// GitHub API docs: https://docs.github.com/en/rest/dependabot/alerts#list-dependabot-alerts-for-an-organization
+func (s *DependabotService) ListOrgAlerts(ctx context.Context, org string, opts *ListAlertsOptions) ([]*DependabotAlert, *Response, error) {
+	url := fmt.Sprintf("orgs/%v/dependabot/alerts", org)
+	return s.listAlerts(ctx, url, opts)
+}
+
+// GetRepoAlert gets a single repository Dependabot alert
+//
+// GitHub API docs: https://docs.github.com/en/rest/dependabot/alerts#get-a-dependabot-alert
+func (s *DependabotService) GetRepoAlert(ctx context.Context, owner string, repo string, number int) (*DependabotAlert, *Response, error) {
```
Comment (gmlewis) [inline github/dependabot_alerts.go:114]:
```suggestion
func (s *DependabotService) GetRepoAlert(ctx context.Context, owner, repo string, number int) (*DependabotAlert, *Response, error) {
```

File: github/dependabot_alerts.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+// ListOrgAlerts lists all Dependabot alerts of an organization
+//
+// GitHub API docs: https://docs.github.com/en/rest/dependabot/alerts#list-dependabot-alerts-for-an-organization
+func (s *DependabotService) ListOrgAlerts(ctx context.Context, org string, opts *ListAlertsOptions) ([]*DependabotAlert, *Response, error) {
+	url := fmt.Sprintf("orgs/%v/dependabot/alerts", org)
+	return s.listAlerts(ctx, url, opts)
+}
+
+// GetRepoAlert gets a single repository Dependabot alert
```
Comment (gmlewis) [inline github/dependabot_alerts.go:111]:
```suggestion
// GetRepoAlert gets a single repository Dependabot alert.
```

File: github/dependabot_alerts.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+// ListRepoAlerts lists all Dependabot alerts of a repository
+//
+// GitHub API docs: https://docs.github.com/en/rest/dependabot/alerts#list-dependabot-alerts-for-a-repository
+func (s *DependabotService) ListRepoAlerts(ctx context.Context, owner, repo string, opts *ListAlertsOptions) ([]*DependabotAlert, *Response, error) {
+	url := fmt.Sprintf("repos/%v/%v/dependabot/alerts", owner, repo)
+	return s.listAlerts(ctx, url, opts)
+}
+
+// ListOrgAlerts lists all Dependabot alerts of an organization
```
Comment (gmlewis) [inline github/dependabot_alerts.go:103]:
```suggestion
// ListOrgAlerts lists all Dependabot alerts of an organization.
```

File: github/dependabot_alerts.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	var alerts []*DependabotAlert
+	resp, err := s.client.Do(ctx, req, &alerts)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return alerts, resp, nil
+}
+
+// ListRepoAlerts lists all Dependabot alerts of a repository
```
Comment (gmlewis) [inline github/dependabot_alerts.go:95]:
```suggestion
// ListRepoAlerts lists all Dependabot alerts of a repository.
```

File: github/dependabot_alerts.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+type ListAlertsOptions struct {
+	State     string `url:"state,omitempty"`
+	Severity  string `url:"severity,omitempty"`
+	Ecosystem string `url:"ecosystem,omitempty"`
+	Package   string `url:"package,omitempty"`
+	Scope     string `url:"scope,omitempty"`
+	Sort      string `url:"sort,omitempty"`
+	Direction string `url:"direction,omitempty"`
```
Comment (gmlewis) [inline github/dependabot_alerts.go:70]:
```suggestion
	State     *string `url:"state,omitempty"`
	Severity  *string `url:"severity,omitempty"`
	Ecosystem *string `url:"ecosystem,omitempty"`
	Package   *string `url:"package,omitempty"`
	Scope     *string `url:"scope,omitempty"`
	Sort      *string `url:"sort,omitempty"`
	Direction *string `url:"direction,omitempty"`
```

File: github/dependabot_alerts.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+	SecurityVulnerability AdvisoryVulnerability      `json:"security_vulnerability"`
+	URL                   string                     `json:"url"`
+	HtmlURL               string                     `json:"html_url"`
+	CreatedAt             Timestamp                  `json:"created_at"`
+	UpdatedAt             Timestamp                  `json:"updated_at"`
+	DismissedAt           *Timestamp                 `json:"dismissed_at"`
+	DismissedBy           *User                      `json:"dismissed_by"`
+	DismissedReason       string                     `json:"dismissed_reason"`
+	DismissedComment      string                     `json:"dismissed_comment"`
+	FixedAt               *Timestamp                 `json:"fixed_at"`
```
Comment (gmlewis) [inline github/dependabot_alerts.go:60]:
```suggestion
	Number                *int                        `json:"number,omitempty"`
	State                 *string                     `json:"state,omitempty"`
	Dependency            *Dependency                 `json:"dependency,omitempty"`
	SecurityAdvisory      *DependabotSecurityAdvisory `json:"security_advisory,omitempty"`
	SecurityVulnerability *AdvisoryVulnerability      `json:"security_vulnerability,omitempty"`
	URL                   *string                     `json:"url,omitempty"`
	HtmlURL               *string                     `json:"html_url,omitempty"`
	CreatedAt             *Timestamp                  `json:"created_at,omitempty"`
	UpdatedAt             *Timestamp                  `json:"updated_at,omitempty"`
	DismissedAt           *Timestamp                 `json:"dismissed_at,omitempty"`
	DismissedBy           *User                      `json:"dismissed_by,omitempty"`
	DismissedReason       *string                     `json:"dismissed_reason,omitempty"`
	DismissedComment      *string                     `json:"dismissed_comment,omitempty"`
	FixedAt               *Timestamp                 `json:"fixed_at,omitempty"`
```

File: github/dependabot_alerts.go [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+	Description     string                  `json:"description"`
+	Vulnerabilities []AdvisoryVulnerability `json:"vulnerabilities"`
+	Severity        string                  `json:"severity"`
+	Cvss            AdvisoryCvss            `json:"cvss"`
+	Cwes            []AdvisoryCwes          `json:"cwes"`
+	Identifiers     []AdvisoryIdentifier    `json:"identifiers"`
+	References      []AdvisoryReference     `json:"references"`
+	PublishedAt     Timestamp               `json:"published_at"`
+	UpdatedAt       Timestamp               `json:"updated_at"`
+	WithdrawnAt     *Timestamp              `json:"withdrawn_at"`
```
Comment (gmlewis) [inline github/dependabot_alerts.go:42]:
```suggestion
	GHSAID          *string                  `json:"ghsa_id,omitempty"`
	CVEID           *string                  `json:"cve_id,omitempty"`
	Summary         *string                  `json:"summary,omitempty"`
	Description     *string                  `json:"description,omitempty"`
	Vulnerabilities []*AdvisoryVulnerability `json:"vulnerabilities,omitempty"`
	Severity        *string                  `json:"severity,omitempty"`
	CVSs            *AdvisoryCvss            `json:"cvss,omitempty"`
	CWEs            []*AdvisoryCwes          `json:"cwes,omitempty"`
	Identifiers     []*AdvisoryIdentifier    `json:"identifiers,omitempty"`
	References      []*AdvisoryReference     `json:"references,omitempty"`
	PublishedAt     *Timestamp               `json:"published_at,omitempty"`
	UpdatedAt       *Timestamp               `json:"updated_at,omitempty"`
	WithdrawnAt     *Timestamp              `json:"withdrawn_at,omitempty"`
```

File: github/dependabot_alerts.go [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+type AdvisoryCvss struct {
+	Score        float64 `json:"score"`
+	VectorString string  `json:"vector_string"`
+}
+
+type AdvisoryCwes struct {
+	CweID string `json:"cwe_id"`
+	Name  string `json:"name"`
```
Comment (gmlewis) [inline github/dependabot_alerts.go:26]:
```suggestion
type AdvisoryCWEs struct {
	CWEID *string `json:"cwe_id,omitempty"`
	Name  *string `json:"name,omitempty"`
```

File: github/dependabot_alerts.go [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+type Dependency struct {
```
Comment (gmlewis) [inline github/dependabot_alerts.go:13]: For this struct and all other exported structs, please add a well-formed Godoc-style comment (complete sentence with punctuation) describing the purpose of the struct and what it represents.

File: github/dependabot_alerts.go [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+	Package      VulnerabilityPackage `json:"package"`
+	ManifestPath string               `json:"manifest_path"`
+	Scope        string               `json:"scope"`
```
Comment (gmlewis) [inline github/dependabot_alerts.go:16]:
```suggestion
	Package      *VulnerabilityPackage `json:"package,omitempty"`
	ManifestPath *string               `json:"manifest_path,omitempty"`
	Scope        *string               `json:"scope,omitempty"`
```

File: github/dependabot_alerts.go [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+type AdvisoryCvss struct {
+	Score        float64 `json:"score"`
+	VectorString string  `json:"vector_string"`
```
Comment (gmlewis) [inline github/dependabot_alerts.go:21]:
```suggestion
	Score        *float64 `json:"score,omitempty"`
	VectorString *string  `json:"vector_string,omitempty"`
```

File: github/github.go [hunk 12]
```diff
@@ -237,6 +237,12 @@ type ListCursorOptions struct {
 	// For paginated result sets, the number of results to include per page.
 	PerPage int `url:"per_page,omitempty"`
 
+	// For paginated result sets, the number of results to include per page starting from the first matching result.
```
Comment (gmlewis) [inline github/github.go:240]:
```suggestion
	// For paginated result sets, the number of results per page (max 100), starting from the first matching result.
	// This parameter must not be used in combination with last.
```

File: github/github.go [hunk 13]
```diff
@@ -237,6 +237,12 @@ type ListCursorOptions struct {
...
+	First int `url:"first,omitempty"`
+
+	// For paginated result sets, the number of results to include per page starting from the last matching result.
```
Comment (gmlewis) [inline github/github.go:243]:
```suggestion
	// For paginated result sets, the number of results per page (max 100), starting from the last matching result.
	// This parameter must not be used in combination with first.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @francois2metz !

**Comment by gmlewis** [comment]
@francois2metz - just FYI - no need to force-push to your PRs in this repo because we always squash-merge at the end anyway, and it is easier for reviewers if you don't. See CONTRIBUTING.md for more details. Also, please make sure to run `gofmt` on the modified files and `go generate ./...` before pushing.  Thanks.

**Review by gmlewis — APPROVED** [review]
Thank you, @francois2metz !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by gmlewis** [comment]
@francois2metz - please make sure to run `gofmt` and `go generate ./...` and push the results to make the tests pass.

**Review by gmlewis — APPROVED** [review]

**Review by valbeat — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**